File: remove_bg.py
from PIL import Image
import os

def remove_checkerboard(image_path, output_path):
    print(f"Processing {image_path}...")
    try:
        img = Image.open(image_path).convert("RGBA")
        datas = img.getdata()

        newData = []
        for item in datas:
            r, g, b, a = item
            
            # The checkerboard is usually composed of pure white (255,255,255) 
            # and a specific light gray. We'll set a threshold for very light colors 
            # that are also grays/whites (where R, G, and B are similar and high).
            
            # Check if color is near white or light gray
            if (r > 200 and g > 200 and b > 200):
                # Calculate color distance to check if it's grayscale-ish
                if abs(r - g) < 20 and abs(r - b) < 20 and abs(g - b) < 20:
                     newData.append((255, 255, 255, 0)) # Make transparent
                else:
                    newData.append(item)
            else:
                newData.append(item)

        img.putdata(newData)
        img.save(output_path, "PNG")
        print(f"Saved cleaned image to {output_path}")
    except Exception as e:
        print(f"Error processing {image_path}: {e}")

# Paths
dir_path = r"c:\Users\pc\Documents\app.g\gmail.app\public"
img1 = os.path.join(dir_path, "email-overload-illustration.png")
img2 = os.path.join(dir_path, "time-saved-illustration.png")

remove_checkerboard(img1, img1)
remove_checkerboard(img2, img2)

File: remove_bg.js
const Jimp = require('jimp');
const path = require('path');

async function removeWhiteBackground(imagePath) {
    try {
        console.log(`Processing ${imagePath}...`);
        const image = await Jimp.read(imagePath);

        image.scan(0, 0, image.bitmap.width, image.bitmap.height, function (x, y, idx) {
            const r = this.bitmap.data[idx + 0];
            const g = this.bitmap.data[idx + 1];
            const b = this.bitmap.data[idx + 2];

            // If the pixel is very light (white or light gray checkerboard pattern)
            if (r > 230 && g > 230 && b > 230) {
                // Check if it's grayscale-ish
                if (Math.abs(r - g) < 15 && Math.abs(r - b) < 15) {
                    this.bitmap.data[idx + 3] = 0; // Make transparent
                }
            }
        });

        await image.writeAsync(imagePath);
        console.log(`Successfully processed ${imagePath}`);
    } catch (error) {
        console.error(`Error processing ${imagePath}:`, error);
    }
}

async function main() {
    const dirPath = path.join(__dirname, 'public');
    const img1 = path.join(dirPath, 'email-overload-illustration.png');
    const img2 = path.join(dirPath, 'time-saved-illustration.png');

    await removeWhiteBackground(img1);
    await removeWhiteBackground(img2);
}

main();
